app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -591,13 +603,38 @@ export default class Heatmap {
 
   sortColumns(direction) {
     this.columnClustering = null;
+    this.gColumnDendogram.classed(cs.shouldSortColumns, true);
```

Comment:
Is the dendogram still present if we are sorting?
This functions should operate on data structures only.. .. the render functons would change anything related to display.

Reply:
The dendrogram is hidden by CSS when sorting. I will add some comments and move those lines to the render functions.

Reply:
It's probably better to just return after `this.gColumnDendogram.select("g").remove();` in the render function. Why draw it?

Reply:
I didn't want to touch the existing rendering because I didn't understand it. (First time working with SVG or D3). So taking your suggestion I will remove the CSS hiding and make the rendering conditional on `this.rowClustering`.